rqt_py_common/src/rqt_py_common/message_helpers.py

```diff
@@ -41,19 +42,123 @@
 SRV_MODE = 'srv'
 ACTION_MODE = 'action'
 
+ROSIDL_FILTERS = {
+    MSG_MODE: lambda n: n.endswith('.msg'),
+    SRV_MODE: lambda n: n.endswith('.srv'),
+    ACTION_MODE: lambda n: n.startswith('action/') and n.endswith('.idl')
```

Comment:
why only filter prefixes for actions?

Reply:
Because the previous logic would collect both ```.msg``` files that were ```srv/*.msg``` and ```msg/*.msg```. Unfortunately there is no rosidl_interface yet that includes an ```.action``` suffix and ```.idl``` is not specific to actions.

Reply:
There are no `.idl` files in the current master branch yet. They are still named `.action`.

Reply:
When I generate the rosidl_interfaces for test_msgs, this is the list that's created and added to the rosidl_interfaces resource_index. I haven't noticed an action/Fibonacci.action being added to this list. Do you know where this happens so I can pull the correct branch and test this out with .action? I've done a basic vcs pull from master, but maybe I'm missing something.

```
action/Fibonacci.idl
action/Fibonacci_Feedback.msg
action/Fibonacci_Goal.srv
action/Fibonacci_Goal_Request.msg
action/Fibonacci_Goal_Response.msg
action/Fibonacci_Result.srv
action/Fibonacci_Result_Request.msg
action/Fibonacci_Result_Response.msg
msg/BoundedArrayNested.idl
msg/BoundedArrayNested.msg
msg/BoundedArrayPrimitives.idl
msg/BoundedArrayPrimitives.msg
msg/BoundedArrayPrimitivesNested.idl
msg/BoundedArrayPrimitivesNested.msg
msg/Builtins.idl
msg/Builtins.msg
msg/DynamicArrayNested.idl
msg/DynamicArrayNested.msg
msg/DynamicArrayPrimitives.idl
msg/DynamicArrayPrimitives.msg
msg/DynamicArrayPrimitivesNested.idl
msg/DynamicArrayPrimitivesNested.msg
msg/DynamicArrayStaticArrayPrimitivesNested.idl
msg/DynamicArrayStaticArrayPrimitivesNested.msg
msg/Empty.idl
msg/Empty.msg
msg/Nested.idl
msg/Nested.msg
msg/Primitives.idl
msg/Primitives.msg
msg/StaticArrayNested.idl
msg/StaticArrayNested.msg
msg/StaticArrayPrimitives.idl
msg/StaticArrayPrimitives.msg
msg/StaticArrayPrimitivesNested.idl
msg/StaticArrayPrimitivesNested.msg
srv/Empty.idl
srv/Empty.srv
srv/Empty_Request.msg
srv/Empty_Response.msg
srv/Primitives.idl
srv/Primitives.srv
srv/Primitives_Request.msg
srv/Primitives_Response.msg
```